lib/teracy-dev-core/config/vmware_provider.rb

```diff
@@ -0,0 +1,46 @@
+require 'teracy-dev/config/configurator'
+
+module TeracyDevCore
+  module Config
+    # see: https://www.vagrantup.com/docs/providers/
+    class VMwareProvider < TeracyDev::Config::Configurator
+
+      def configure_node(settings, config)
+        providers_settings = settings['providers'] ||= []
+
+        providers_settings.each do |provider_settings|
+          next if !provider_settings['enabled']
+          if provider_settings['type'].include?('vmware')
+            @logger.debug("provider_settings: #{provider_settings}")
+            configure_vmware(provider_settings, config)
+          end
+        end
+      end
+
+      private
+
+      def configure_vmware(provider_settings, config)
+        config.vm.provider provider_settings['type'] do |vmware|
+          vmware.gui = true if provider_settings['gui'] == true
+
+          provider_settings.each do |key, val|
+            if provider_settings[key] == "vmx" and !provider_settings[key].nil?
+              provider_settings['vmx'].each do |vmx_key, vmx_val|
+                vmx_val = vmx_val.to_s.strip()
+                if !vmx_val.empty?
+                  vmware.vmx["#{vmx_key}"] = vmx_val
+                end
+              end
+            else
+              val = val.to_s.strip()
+              if !val.empty?
+                vmware.provider_settings[key] = val
```

Comment:
the setting like (dynamic key names) this never works, try with other public plugin configs to see:
- https://github.com/teracyhq-incubator/teracy-dev-core/search?q=set_options&unscoped_q=set_options
- https://github.com/hashicorp/vagrant/blob/master/lib/vagrant/plugin/v2/config.rb